airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -73,6 +73,17 @@ public <T> Stream<T> unsafeResultSetQuery(final CheckedFunction<Connection, Resu
         });
   }
 
+  /**
+   * Implementations of DatabaseMetadata hold a reference of the Connection object. It is safe to use this to retrieve
```

Comment:
maybe worth adding a `DANGER!!!` or something, to make this more obvious

idk if it's correct to deprecate though, since e.g. getIdentifierQuoteString is a totally valid usecase :/